src/cli.js

```diff
@@ -1,14 +1,64 @@
 #!/usr/bin/env node
+let chalk = require('chalk')
+let hydrate = require('@architect/hydrate')
+let http = require('./http')
+let path = require('path')
 let sandbox = require('./index')
 let ver = require('../package.json').version
 let version = `Sandbox ${ver}`
+let watch = require('node-watch')
 let options = process.argv
-sandbox.start({version, options}, function _done(err) {
-  if (err && err.message === 'hydration_error') {
-    process.exit(1)
-  }
-  if (err) {
-    console.log(err)
-    process.exit(1)
-  }
-})
+
+start()
+function start() {
+  sandbox.start({version, options}, function watching(err) {
+    if (err && err.message === 'hydration_error') {
+      // Hydration errors already reported, no need to log
+      process.exit(1)
+    }
+    if (err) {
+      console.log(err)
+      process.exit(1)
+    }
+    let watcher = watch(process.cwd(), { recursive: true })
+    let arcFile = new RegExp(`${process.cwd()}${path.sep}(\\.arc|app\\.arc|arc\\.yaml|arc\\.json)`)
+
+    watcher.on('change', function(event, name) {
+      if (event === 'update' &&
+          name.includes(`${process.cwd()}/src/shared`) ||
+          name.includes(`${process.cwd()}/src/views`)) {
+        let indicator = chalk.green.dim('⚬')
+        let status = chalk.grey('Shared file changed, rehydrating functions...')
+        console.log(`${indicator} ${status}`)
+
+        let start = Date.now()
+        hydrate.shared(() => {
+          let indicator = chalk.green.dim('✓')
+          let end = Date.now()
+          let status = chalk.grey(`Project files rehydrated into functions in ${end - start}ms`)
+          console.log(`${indicator} ${status}`)
+        })
+      }
+      if (event === 'update' &&
+          name.match(arcFile)) {
+        let indicator = chalk.green.dim('⚬')
+        let status = chalk.grey('Architect project manifest changed, reloading HTTP routes...')
+        console.log(`${indicator} ${status}`)
+
+        let start = Date.now()
+        process.env.QUIET = true
+        http.close()
+        http.start(function () {
```

Comment:
From an abstraction point of view, I would be weary of mixing using the `sandbox.start` and `http.start` / `http.close` APIs in here. If `sandbox.start` changes how it uses `http`, but we don't update the usage of `http` in this file, then we could get our streams crossed up. I would consider modifying or adding to the `sandbox.start` API to enable the kind of interactions you are looking for here, _or_, restarting the entire sandbox using the `end()` function provided to the `start` callback, and calling up `sandbox.start` again.

Reply:
Yeah, as a separate exercise, related to this work we should probably establish a cleaner API into the various services sandbox starts and closes.